### ISSUE TITLE
ability to do comptime @ptrToInt on ConstPtrSpecialRef

### ISSUE THREAD

**Issue by shawnl** [ziglang/zig#3021]
In my SIMD work I am running across a need to be able to do pointer arithmetic on vectors, something like this:

```
var table: [54]u6 = ...;
var indexes: @Vector(4, u32) = [_]{1, 2, 3, 4};
var pointers_to_indexes: @vector(4, *u32) = [_]*u32{&table[indexes[0]], &table[indexes[1]], &table[indexes[2]], &table[indexes[3]]}; <== but in parallel.
```

This can be done with `@ptrToInt`, however `@ptrToInt` does not work comptime on references.

This is actually a kinda tricky bug to fix, and is thus is mainly a nice-to-have, but not urgent.

**Comment by shawnl** [comment]
Ahhhh, the problem is that we are not using ConstPtrSpecialBaseArray
